block-storage/src/main/scala/coop/rchain/blockstorage/BlockDagFileStorage.scala

```diff
@@ -897,12 +947,33 @@ object BlockDagFileStorage {
                }
     } yield result
 
-  def create[F[_]: Concurrent: Sync: Log: BlockStore](
+  private def loadBlockNumberIndexLmdbDbi[F[_]: Sync: Log: RaiseIOError](
+      config: Config
+  ): F[LmdbDbi[F, ByteBuffer]] =
+    for {
+      notExists <- notExists[F](config.blockNumberIndexPath)
+      _         <- if (notExists) makeDirectory[F](config.blockNumberIndexPath) else ().pure[F]
+      env <- Sync[F].delay {
+              val flags = if (config.noTls) List(EnvFlags.MDB_NOTLS) else List.empty
+              Env
+                .create()
+                .setMapSize(config.mapSize)
+                .setMaxDbs(config.maxDbs)
+                .setMaxReaders(config.maxReaders)
```

Comment:
food for thought: this is aligned with what rspace needs. might be misaligned for blockstore

Reply:
Consider

```
    for {
      _ <- notExists[F](config.blockNumberIndexPath).ifM(
                        makeDirectory[F](config.blockNumberIndexPath), 
                        ().pure[F]
                 )
```